### PR TITLE
[RFR] VersionPick resolution in arguments

### PR DESCRIPTION
**Pull request by izapolsk**
This PR allows to use VersionPick in any widget method. Additionally, it fixed VersionPick resolution issues in __init__ when widget parses arguments before passing them to parent (like issue with BootstrapSwitch).

### CODE REVIEW

File: src/widgetastic/widget.py [hunk 1]
```diff
@@ -209,8 +209,9 @@ def __new__(cls, name, bases, attrs):
                 # handle read() specifics
                 new_attrs[key] = logged(log_result=True)(value)
             else:
-                # Do nothing
-                new_attrs[key] = value
+                # VersionPick resolution wrapper
+                new_attrs[key] = resolve_version_pick(key)(value)
```
Comment (mfalesni) [inline src/widgetastic/widget.py]: How do you detect that it is a method? What about class attributes? You probably need to check if it is a callable, or a function ... depends how exactly the class machinery wraps the function as a method :)
We also need to implement a blacklist for this feature, so you can omit methods on purpose.

File: src/widgetastic/utils.py [hunk 2]
```diff
@@ -610,3 +611,18 @@ def __get__(self, o, t):
...
+        version = func.browser.product_version
+        wargs = []
+        for arg in args:
+            wargs.append(arg.pick(version) if isinstance(arg, VersionPick) else arg)
+        wkwargs = kwargs.copy()
+        for arg, value in wkwargs.items():
+            wkwargs[arg] = arg.pick(version) if isinstance(arg, VersionPick) else arg
+        r = func(*wargs, **wkwargs)
+        return r
+    return wrapped
```
Comment (mfalesni) [inline src/widgetastic/utils.py]: You don't need to write this function as it is. Use `ConstructorResolvable` class instead. Like in src/widgetastic/widget.py#L36

**Comment by izapolsk** [comment]
@mfalesni, I was writing this PR about 12 pm yesterday. I even haven't test yet what I wrote :) I'll take into account all your suggessions.

**Comment by mfalesni** [comment]
@izapolsk Sure, no prob, just wanted to review early :)

**Comment by psav** [comment]
I'm happy but I'll let mfalesni merge

**Comment by izapolsk** [comment]
@mfalesni, please review when have time.

**Pull request merged**